File: app/src/main/java/my/com/toru/multiviewbinder/main/data/MainProductData.java
package my.com.toru.multiviewbinder.main.data;

import my.com.toru.multiviewbinder.uicomponent.ListItemType;

public final class MainProductData implements ListItemType{
    public static final int TYPE = 1;

    private String name;
    private int count;

    public MainProductData(String name, int count) {
        this.name = name;
        this.count = count;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getCount() {
        return count;
    }

    public void setCount(int count) {
        this.count = count;
    }

    @Override
    public int getListItemType() {
        return ListItemType.TYPE_PRODUCT;
    }
}

File: app/src/main/java/my/com/toru/multiviewbinder/main/data/MainProductData.kt
package my.com.toru.multiviewbinder.main.data

import my.com.toru.multiviewbinder.uicomponent.ListItemType

class MainProductData(var name: String?, var count: Int) : ListItemType {

    override fun getListItemType(): Int {
        return ListItemType.TYPE_PRODUCT
    }

    companion object {
        val TYPE = 1
    }
}